graphql-ballerina/duplicate_field_remover.bal

```diff
@@ -0,0 +1,194 @@
+        string[] visitedFields = [];
+        parser:FieldNode[] duplicateFields = [];
+        foreach parser:FieldNode fieldNode in fields {
+            if (visitedFields.indexOf(fieldNode.getName()) == ()) {
+                visitedFields.push(fieldNode.getName());
+            } else {
+                duplicateFields.push(fieldNode);
+            }
+        }
+        int i = 0;
+        while (i < fields.length()) {
+            foreach parser:FieldNode duplicate in  duplicateFields {
+                if (fields[i] === duplicate) {
+                    parser:FieldNode remove = fields.remove(i);
+                    i -= 1;
+                }
+            }
+            i += 1;
+        }
+    }
+
+    private isolated function removeDuplicateFragments(string[] fragments) {
+        string[] visitedFragmentOnTypes = [];
+        string[] duplicateFragments = [];
+        foreach string fragment in fragments {
+            parser:FragmentNode fragmentNode = self.documentNode.getFragments().get(fragment);
+            if (visitedFragmentOnTypes.indexOf(fragmentNode.getOnType()) == ()) {
+                visitedFragmentOnTypes.push(fragmentNode.getOnType());
+            } else {
+                if (!fragmentNode.isInlineFragment()) {
+                   parser:FragmentNode removed = self.documentNode.getFragments().remove(fragment);
+                }
+                duplicateFragments.push(fragment);
+            }
+        }
+        foreach string duplicate in duplicateFragments {
+            string removed = fragments.remove(<int>fragments.indexOf(duplicate));
+        }
+    }
+
+    private isolated function appendDuplicateSelections(parser:Selection duplicate, parser:Selection original) {
+        if (duplicate.isFragment) {
+            parser:FragmentNode duplicateFragmentNode = self.documentNode.getFragments().get(duplicate.name);
+            parser:FragmentNode originalFragmentNode = self.documentNode.getFragments().get(original.name);
+            self.appendDuplicateFragments(duplicateFragmentNode, originalFragmentNode);
+        } else {
+            self.appendDuplicateFields(<parser:FieldNode>duplicate?.node, <parser:FieldNode>original?.node);
+        }
+    }
+
+    private isolated function appendDuplicateFields(parser:FieldNode duplicate, parser:FieldNode original) {
+        foreach var fields in duplicate.getFields() {
+            original.addField(fields);
+        }
+        foreach var fragments in duplicate.getFragments() {
+            original.addFragment(fragments);
+        }
+        foreach var selections in duplicate.getSelections() {
+            original.addSelection(selections);
+        }
+        foreach var arguments in duplicate.getArguments() {
+            original.addArgument(arguments);
+        }
+    }
+
+    private isolated function appendDuplicateFragments(parser:FragmentNode duplicate, parser:FragmentNode original) {
+        foreach var fields in duplicate.getFields() {
```

Comment:
In these places it is better to use the exact type instead of `var`, since we do know the type of the element. Check the other places as well.